pdf_ocr.go

```diff
@@ -0,0 +1,163 @@
+// +build ocr
+
+package docconv
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"sync"
+)
+
+var (
+	exts = []string{".jpg", ".tif", ".tiff", ".png", ".pbm"}
+)
+
+func compareExt(ext string, exts []string) bool {
+	for _, e := range exts {
+		if ext == e {
+			return true
+		}
+	}
+	return false
+}
+
+func cleanupTemp(tmpDir string) {
+	err := os.RemoveAll(tmpDir)
+	if err != nil {
+		log.Println(err)
+	}
+}
+
+func ConvertPDFImages(path string) (BodyResult, error) {
+	bodyResult := BodyResult{}
+
+	tmp, err := ioutil.TempDir("/tmp", "tmp-imgs-")
+	if err != nil {
+		bodyResult.err = err
+		return bodyResult, err
+	}
+	tmpDir := fmt.Sprintf("%s/", tmp)
+
+	defer cleanupTemp(tmpDir)
+
+	_, err = exec.Command("pdfimages", "-j", path, tmpDir).Output()
+	if err != nil {
+		return bodyResult, err
+	}
+
+	files := []string{}
+
+	walkFunc := func(path string, info os.FileInfo, err error) error {
+		path, err = filepath.Abs(path)
+		if err != nil {
+			return err
+		}
+
+		if compareExt(filepath.Ext(path), exts) {
+			files = append(files, path)
+		}
+		return nil
+	}
+	filepath.Walk(tmpDir, walkFunc)
+
+	if len(files) < 1 {
+		return bodyResult, nil
+	}
+
+	var wg sync.WaitGroup
+
+	data := make(chan string, 1)
+
+	wg.Add(len(files))
+	for _, p := range files {
+		go func(pathFile string) {
+
+			f, err := os.Open(pathFile)
+
+			if err != nil {
+				bodyResult.err = err
+			}
+
+			defer f.Close()
```

Comment:
If the file failed to open, you can't close it, this will panic. Need an early return